pyiron/lammps/control.py

```diff
@@ -270,6 +257,16 @@ def pressure_to_lammps(self, pressure, rotation_matrix):
                  else p)
                 for p in pressure]
 
+    @staticmethod
+    def _is_isotropic_hydrostatic(pressure):
+        return (
+            None not in pressure[:3] and np.allclose(pressure[:3], pressure[0])
+            and (
+                all(p is None for p in pressure[3:])
+                or (None not in pressure[3:] and np.allclose(pressure[3:], 0.))
```

Comment:
I find it still difficult to understand. I'd like to propose two possibilities:

1. We simply say it's an input and whoever puts 1.0e-16 instead of 0 must know what they're doing -> Instead of checking whether the values are close to 0, we check whether it's a clear 0 or not
2. We include one more step before `return` to round up `pressure`.

What do you think?

Reply:
In both cases we'd have `all(p==0 for p in pressure[3:])` instead of `None not in pressure[3:] and np.allclose(pressure[3:], 0.)`

Reply:
@samwaseda ok, I have no objection to breaking it down further to improve readability, which is what I interpret (2) to be.

I'm hard rejecting (1). The only way comparing to a clear 0 is fair is if we cast what we're comparing as an int, and that sounds super sketchy. Plus it doesn't help us for the isotropic evaluation where we need to make sure the first three components compare equally to each other. Anyhow, when I get a few minutes I'll do (2) and hopefully remove the desire for (1) ;)

Reply:
And/or we can maybe separate the two parts:

‘’’
if (None in pressure[:3] or not np.allclose(pressure[:3], pressure[0])):
    return False
if ...
‘’’

Again this is just a suggestion. I don’t strongly stick to it.

Reply:
(Sorry I’m writing it on my phone I know it looks horrible)